kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -980,28 +984,44 @@ def _initialize_service(
             )
 
 
-def _error_message(error: HTTPError) -> str:
-    def format_message(msg):
-        return "".join([msg[0].lower(), msg[1:-1], msg[-1] if msg[-1] != "." else ""])
+def format_message(msg):
+    return "".join([msg[0].lower(), msg[1:-1], msg[-1] if msg[-1] != "." else ""])
 
+def _error_message(error: HTTPError) -> str:
     try:
-        error_json = error.response.json()
-        messages = []
-        reason = error_json.get("reason", None)
-        details = error_json.get("details", None)
-        if reason:
-            messages.append(format_message(reason))
-        if details:
-            messages.append(format_message(details))
-        messages = messages if reason or details else [str(error)]
-        return ". ".join(messages)
+        error_response = error.response
+        return _error_from_response(error_response)
     except Exception as e:
         pass
     try:
         return format_message(error.response.text())
     except Exception:
         return format_message(str(error))
 
+def _error_from_response(response: Response) -> str:
+    messages = []
+    error_json = response.json()
+    reason = error_json.get("reason", None)
+    details = error_json.get("details", None)
+    if reason:
+        messages.append(reason)
+    if details:
+        result = details.get('result', None)
+        if result:
+            the_details = result.get("detail", None)
+            if the_details:
+                if isinstance(the_details, list):
+                    for detail in the_details:
+                        messages.append(f"Reason: {detail['resultMessage']} "\
+                                        f"for the shape: {detail['sourceShape']}")
+                elif isinstance(the_details, dict):
+                    messages.append(f"Reason: {detail['resultMessage']} "\
+                                     f"for the shape: {detail['sourceShape']}")
+                else:
+                    messages.append(str(the_details))
+            else:
+                messages.append(str(result))
+    return ". ".join(messages)
```

Comment:
Is it an absolute must to join these? Or, better yet, does the result dictionaries have to be converted to `str`?

Reply:
This is a choice, would you like to have the full schacl error result? Maybe we can give it back when the debug option is true. However, if the intention is to make a nicer Error log, we can add it here, I don't know, what do you think?

Reply:
No, no. Not the full shacl output, absolutely not. It's just for flexibility of further handling the output.
For example, if the dict you gather above is kept as a dict, it would allow to do something like:
```python
try:
    reason = "Flux Capacitor is broken"
    troubleshooting = {"attention": "Disconnect capacitor drive before opening", "safety": "Shield eyes from light", "actions": "do magic"}
    raise ValueError(reason, troubleshooting)
except ValueError as e:
    print(f"ERROR: {e.args[0]}")
    print("\nTroubleshooting Tips:")
    for k,v in e.args[1].items():
        print(f"\t{k.ljust(10)}: {v}")
```
OUTPUT:
```
ERROR: Flux Capacitor is broken

Troubleshooting Tips:
        attention : Disconnect capacitor drive before opening
        safety    : Shield eyes from light
        actions   : do magic
```
or we can just leave it as it is:
```python
raise ValueError(reason, troubleshooting)
```
OUTPUT:
```
# <traceback>
ValueError: ('Flux Capacitor is broken', {'attention': 'Disconnect capacitor drive before opening', 'safety': 'Shield eyes from light', 'actions': 'do magic'})
```

Reply:
On nexus-forge side, you would still able to format the messages any way you want in `execution.py`. What do you think?

Reply:
Ah ok ok, I understand now your point, yes, I like it, I will change the printing to use the full dict